src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/client/NonAssociationsApiClient.kt

```diff
@@ -19,22 +19,23 @@ class NonAssociationsApiClient(
 
   companion object {
     val LOG: Logger = LoggerFactory.getLogger(this::class.java)
-    private val TYPE_FOR_OFFENDER_NONASSOCIATION = object : ParameterizedTypeReference<OffenderNonAssociationDetailsDto>() {}
+    private val TYPE_FOR_OFFENDER_NONASSOCIATION = object : ParameterizedTypeReference<PrisonerNonAssociationDetailsDto>() {}
   }
 
-  fun getOffenderNonAssociation(offenderNo: String): OffenderNonAssociationDetailsDto? {
-    LOG.debug("Entered getOffenderNonAssociation $offenderNo")
+  fun getPrisonerNonAssociation(prisonerNumber: String): PrisonerNonAssociationDetailsDto? {
+    LOG.debug("Entered getPrisonerNonAssociation $prisonerNumber")
+
     return webClient.get()
-      .uri("/legacy/api/offenders/$offenderNo/non-association-details?currentPrisonOnly=true&excludeInactive=true")
+      .uri("/prisoner/$prisonerNumber/non-associations?includeOtherPrisons=true")
```

Comment:
should includeOtherPrisons be false or will it not matter as we do a prison check later?

Reply:
includeOtherPrisons not sure really,... could be false,.. just was a little concerned if a prisoner had been transferred temporarily but the visit is later on,...